### PR TITLE
right menu bar

### PR DESCRIPTION
**Pull request by romaGri**
#35 closed
Screenshot_1583749645

### CODE REVIEW

**Comment by fryette** [comment]
please attach screenshot

File: ecommers/lib/ui/widgets/right_menu_bar/right_menu_item.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          const SizedBox(width: Insets.x2_5),
+          SvgPicture.asset(Assets.menuArrowIcon),
+        ],
+      ),
+    );
+  }
+}
+
+@override
+Widget _subTitleWidget(RightMenuItemModel model) {
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/right_menu_bar/right_menu_item.dart]: Consider removing redundant method

File: ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  final RightMenuItemModel model;
+
+  const RightMenuSubTitle({@required this.model});
+
+  @override
+  Widget build(BuildContext context) {
+    final modelItem = model.runtimeType;
+    switch (modelItem) {
+      case RightMenuSubTitleModel:
+        return createSubTitle(context, model as RightMenuSubTitleModel);
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart]: Consider moving such logic 'model as RightMenuSubTitleModel' inside method directly

Comment (amarchenko25) [reply ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart]: and the same for other 'create' methods

File: ecommers/lib/extensions/string_extension.dart [hunk 3]
```diff
@@ -1,3 +1,12 @@
+import 'package:flutter/widgets.dart';
+
 extension StringExtension on String {
   bool get isNullOrEmpty => this == null || isEmpty;
-}
\ No newline at end of file
+
+  Color fromHex() {
```
Comment (amarchenko25) [inline ecommers/lib/extensions/string_extension.dart]: rename to fromHexToColor()

File: ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart [hunk 4]
```diff
@@ -14,21 +14,22 @@ class RightMenuSubTitle extends StatelessWidget {
     final modelItem = model.runtimeType;
     switch (modelItem) {
       case RightMenuSubTitleModel:
-        return createSubTitle(context, model as RightMenuSubTitleModel);
+        return createSubTitle(context, model);
```
Comment (amarchenko25) [inline ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart]: Model argument no longer needed.
Please recheck other 'create' methods

**Review by amarchenko25 — DISMISSED** [review]

**Review by amarchenko25 — APPROVED** [review]

**Pull request merged**